### ISSUE TITLE
Intellisense problems with requires, concept and std::vector with g++ 13

### ISSUE THREAD

**Issue by ebaumotte** [microsoft/vscode-cpptools#10925]
### Environment

- OS and Version: Fedora 38
- VS Code Version: 1.78.0
- C/C++ Extension Version: 1.15.4

### Bug Summary and Steps to Reproduce

Bug Summary:

Offending code example:
```
template <typename T>
concept VectorType = requires(T) { typename T::value_type; } && std::convertible_to<T, std::vector<typename T::value_type>>;

template <typename T>
concept OtherConcept = requires(T a) { a._a; };

template <typename T>
    requires std::integral<T>
void foo(T val) {
    std::cout << "a: " << val << "\n";
}

template <typename T>
    requires VectorType<T>
void foo(T val) {
    for (const auto& v : val)  {
        std::cout << "a: " << v << "\n";
    }
}

template <typename T>
    requires OtherConcept<T>
void foo(T val) {
    std::cout << "a: " << val._a << "\n";
}

// S fulfils other concept - this works fine (so the problem is not with concepts in general, just the std::vector one)
struct S {
    int _a;
};

static_assert(OtherConcept<S>); // all fine, as it should be

int main() {
    foo(10); // no issue

    /*
        reports:
        no instance of overloaded function "foo" matches the argument listC/C++(304)
        test_example.cpp(32, 5): argument types are: (std::vector<int, std::allocator<int>>)

        that is incorrect, it compiles fine under gcc 13.1.1 and clang++ 16.0.2
    */
    foo(std::vector<int>{10}); // this gets squiggles

    foo(S{10}); // no issue (as it should be) even though it also uses concepts/requires
}
```

TLDR:
Error squiggles + message appear whenever functions get used that require a concept that has (as far as I can tell) anything to do with an std::vector, even if the code compiles correctly. 

### Configuration and Logs

```shell
Config:

Logs:
C/C++: Log Diagnostics :

-------- Diagnostics - 6.5.2023, 14:21:34
Version: 1.15.4
Current Configuration:
{
    "name": "Linux",
    "includePath": [
        ... project include
    ],
    "defines": [],
    "compilerPath": "/usr/bin/clang",
    "cStandard": "c17",
    "cppStandard": "c++17",
    "intelliSenseMode": "linux-clang-x64",
    "compilerPathInCppPropertiesJson": "/usr/bin/clang",
    "intelliSenseModeIsExplicit": false,
    "cStandardIsExplicit": false,
    "cppStandardIsExplicit": false,
    "mergeConfigurations": false,
    "compilerPathIsExplicit": false,
    "configurationProvider": "ms-vscode.cmake-tools",
    "browse": {
        "path": [
            "${workspaceFolder}/**"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
Custom configurations: 
[ file.cpp ]
{
    "includePath": [
        ... (project includes)
    ],
    "defines": [
        ... (project variables)
    ],
    "compilerPath": "/usr/bin/g++",
    "compilerArgs": [],
    "compilerFragments": [
        " -fPIC -fPIC -fext-numeric-literals -Wall -pthread -Wno-array-bounds -Wno-stringop-overflow  -Wextra -pedantic -Werror -pedantic-errors -O3 -DNDEBUG -std=gnu++20"
    ]
}
```
```
Process ID: 2252352
    Memory Usage: 521 MB
    Compiler Path: /usr/bin/g++
    Includes:
        ... project includes
        /usr/include/c++/13
        /usr/include/c++/13/x86_64-redhat-linux
        /usr/include/c++/13/backward
        /usr/lib/gcc/x86_64-redhat-linux/13/include
        /usr/local/include
        /usr/include
    Defines:
        ... project defines
        NDEBUG
    Standard Version: c++20
    IntelliSense Mode: linux-gcc-x64
    Other Flags:
        --g++
        --gnu_version=130101
Total Memory Usage: 1565 MB
```
```


### Other Extensions

_No response_

### Additional context

_No response_

**Comment by ebaumotte** [comment]
Lmk if there is anything else I can provide to fix this.

**Comment by sean-mcmanus** [comment]
The repro requires
```
#include <vector>
#include <iostream>
```
to be added.

It also requires g++ 13 and doesn't repro with earlier versions.

**Comment by sean-mcmanus** [comment]
As a potential workaround, you could set compilerPath to a g++-12 so that IntelliSense will use those headers instead and you could still compile your code with g++-13.

**Comment by Colengms** [comment]
Given that the code example uses `std::convertible_to`, this is most likely the same issue as: https://github.com/microsoft/vscode-cpptools/issues/11038

... which appears to be (arguably) a gcc-13 header bug, as it's using a built-in without checking for support for that built-in, as the system headers do elsewhere.

Indeed, bypassing the check against `std::convertible_to`, does remove the squiggle.

**Comment by Colengms** [comment]
Using https://github.com/microsoft/vscode-cpptools/issues/11038 to track this.  Closing this one as a duplicate.

**Comment by michelleangela** [comment]
Fix available in pre-release https://github.com/microsoft/vscode-cpptools/releases/tag/v1.17.0.
